src/lib/frontend/models.ml

```diff
@@ -481,6 +481,11 @@ let pp_constant ppf (_, t) =
   Format.fprintf ppf "%a" SmtlibCounterExample.pp_dummy_value_of_type t
 
 let output_concrete_model fmt props ~functions ~constants ~arrays =
+  if ModelMap.(is_suspicious_model functions || is_suspicious_model constants
+               || is_suspicious_model arrays) then
+    Format.fprintf fmt "This model is a best-effort. Please note tht it \
+                        could be wrong. @.";
```

Comment:
Not a big fan of the formulation as it seems to imply that when the message is not there we claim that the model is correct.

What about something more along the lines of "Note: this model includes symbols for which model generation is known to be incomplete." which is more precise and also doesn't promise anything.

(BTW the model needs to be parseable, so this should be a Lisp comment i.e. the line must start with `;`)